src/Microsoft.Security.Utilities.Core/IdentifiableScan.cs

```diff
@@ -315,30 +360,37 @@ public IEnumerable<Detection> DetectSecrets(Stream file)
                 {
                     foreach (string signature in signatures)
                     {
-                        string found = System.Text.Encoding.UTF8.GetString(text, 0, (int)textLength);
+                        string found = Encoding.UTF8.GetString(text, 0, (int)textLength);
 
                         if (found.IndexOf(signature) != -1 &&
                             signatureToPatternsMap.TryGetValue(signature, out IList<RegexPattern> patterns))
                         {
                             foreach (RegexPattern pattern in patterns)
                             {
-                                string redactionToken = null;
-
-                                if (generateCorrelatingIds)
-                                {
-                                    redactionToken = RegexPattern.GenerateCrossCompanyCorrelatingId(found);
-                                }
-
                                 var result = pattern.GetMatchIdAndName(found);
 
                                 if (result != null)
                                 {
+                                    string c3id = null;
+                                    string preciseId = result.Item1;
+
+                                    if (generateCorrelatingIds)
+                                    {
+                                        c3id = RegexPattern.GenerateCrossCompanyCorrelatingId(found);
+                                    }
+
+                                    string redactionToken = c3id != null 
+                                        ? $"{preciseId}:{c3id}"
+                                        : RegexPattern.FallbackRedactionToken;
+
                                     yield return new Detection
                                     {
-                                        Id = result.Item1,
+                                        Id = preciseId,
                                         Name = result.Item2,
                                         Start = (int)start,
                                         Length = (int)textLength,
+                                        Metadata = DetectionMetadata.HighEntropy,
```

Comment:
>HighEntropy [](http://example.com/codeflow?start=69&length=11)

All rules handled by the high perf scanner are high entropy. Mostly these checks are identifiable, we need to update this metadata to be more of a first class property (bundled with signatures, passed to Rust).